### PR TITLE
Rancher 642 Roll out sorry-cypress for cypress tests run

### PR DESCRIPTION
**Pull request by vitalii-ustalu**

### CODE REVIEW

**Review by vkapylou — APPROVED** [review]

**Review by renatsafiulin — APPROVED** [review]

File: pipelines/rancher/cluster.groovy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!groovy
-@Library('pipelines-shared-library') _
+@Library('pipelines-shared-library@RANCHER-642') _
```
Comment (OHaimanov) [inline pipelines/rancher/cluster.groovy:2]: Do not forget to remove branch before merge

File: pipelines/tests/cypress/cypressTestsPipeline.groovy [hunk 2]
```diff
@@ -2,7 +2,7 @@ import org.folio.Constants
 import org.jenkinsci.plugins.workflow.libs.Library
 import hudson.util.Secret
 
-@Library('pipelines-shared-library') _
+@Library('pipelines-shared-library@RANCHER-642') _
```
Comment (OHaimanov) [inline pipelines/tests/cypress/cypressTestsPipeline.groovy:5]: Same here

File: pipelines/tests/cypress/cypressTestsScheduledPipeline.groovy [hunk 3]
```diff
@@ -1,12 +1,12 @@
-@Library('pipelines-shared-library') _
+@Library('pipelines-shared-library@RANCHER-642') _
```
Comment (OHaimanov) [inline pipelines/tests/cypress/cypressTestsScheduledPipeline.groovy:1]: Also do not forget to remove branch before merge

File: vars/cypressStages.groovy [hunk 4]
```diff
@@ -2,105 +2,131 @@ import org.folio.Constants
 import org.jenkinsci.plugins.workflow.libs.Library
 import groovy.transform.Field
 
-@Library('pipelines-shared-library') _
+@Library('pipelines-shared-library@RANCHER-642') _
```
Comment (OHaimanov) [inline vars/cypressStages.groovy:5]: Do not forget to remove branch before merge

File: pipelines/tests/cypress/cypressTestsPipeline.groovy [hunk 5]
```diff
@@ -50,128 +46,10 @@ properties([
...
-                script {
-                    zip zipFile: "allure-results.zip", glob: "allure-results/*"
-
-                    archiveArtifacts allowEmptyArchive: true, artifacts: "allure-results.zip", fingerprint: true, defaultExcludes: false
-                }
-            }
-        }
-    }
+node {
+    cypressStages(params)
```
Comment (OHaimanov) [inline pipelines/tests/cypress/cypressTestsPipeline.groovy:54]: What do you think about cypressFlow?

File: vars/cypressStages.groovy [hunk 6]
```diff
@@ -2,105 +2,131 @@ import org.folio.Constants
...
-            sh "yarn install"
-        }
-    }
+    def customBuildName = params.customBuildName?.trim() ? params.customBuildName + '.' + env.BUILD_ID : env.BUILD_ID
+    buildName customBuildName
+    def cypressWorkers = [:]
+    int numberOfWorkers = params.numberOfWorkers as int ?: 1
+    def resultPaths = []
+    for (int workerNumber = 1; workerNumber <= numberOfWorkers; workerNumber++) {
+        cypressWorkers["CypressWorker№${workerNumber}"] = {
```
Comment (OHaimanov) [inline vars/cypressStages.groovy:16]: I think it is better to avoid № usage better #

**Review by OHaimanov — APPROVED** [review]

**Review by kdubniak — APPROVED** [review]

**Pull request merged**